webknossos-datastore/app/com/scalableminds/webknossos/datastore/datavault/HttpsDataVault.scala

```diff
@@ -0,0 +1,104 @@
+package com.scalableminds.webknossos.datastore.datavault
+
+import com.scalableminds.webknossos.datastore.storage.{
+  FileSystemCredential,
+  HttpBasicAuthCredential,
+  LegacyFileSystemCredential,
+  RemoteSourceDescriptor
+}
+
+import java.net.URI
+import scala.concurrent.duration.DurationInt
+import sttp.client3._
+import sttp.model.Uri
+
+class HttpsDataVault(credential: Option[FileSystemCredential]) extends DataVault {
+
+  private val connectionTimeout = 5 seconds
+  private val readTimeout = 1 minute
+  private lazy val backend = HttpClientSyncBackend(options = SttpBackendOptions.connectionTimeout(connectionTimeout))
+
+  def getBasicAuthCredential: Option[HttpBasicAuthCredential] =
+    credential match {
+      case Some(c) => {
```

Comment:
```suggestion
    credential.map{ c =>
```

Reply:
Doesn't work for me, since the inner match may return none which is not possible in the map which expects a full result.

Reply:
maybe flatMap, then? 🤔 Feel free to ignore if that is also not a good fit

Reply:
Yes flatMap it is!